src/main/java/wooteco/subway/admin/service/LineService.java

```diff
@@ -3,52 +3,107 @@
 import org.springframework.stereotype.Service;
 import wooteco.subway.admin.domain.Line;
 import wooteco.subway.admin.domain.LineStation;
-import wooteco.subway.admin.domain.Station;
 import wooteco.subway.admin.dto.LineResponse;
 import wooteco.subway.admin.dto.LineStationCreateRequest;
+import wooteco.subway.admin.dto.StationResponse;
 import wooteco.subway.admin.repository.LineRepository;
+import wooteco.subway.admin.repository.LineStationRepository;
 import wooteco.subway.admin.repository.StationRepository;
 
+import java.util.LinkedHashSet;
 import java.util.List;
-import java.util.Set;
+import java.util.stream.Collectors;
 
+@Service
 public class LineService {
+    private static final String ERROR_MESSAGE_NAME_OVER_LAP = "존재하는 이름입니다";
+    private static final String ERROR_MESSAGE_NOT_EXIST_LINE_ID = "해당 아이디의 노선이 존재하지 않습니다";
+
     private LineRepository lineRepository;
+    private LineStationRepository lineStationRepository;
     private StationRepository stationRepository;
 
-    public LineService(LineRepository lineRepository, StationRepository stationRepository) {
+    public LineService(LineRepository lineRepository,
+                       LineStationRepository lineStationRepository,
+                       StationRepository stationRepository) {
         this.lineRepository = lineRepository;
+        this.lineStationRepository = lineStationRepository;
         this.stationRepository = stationRepository;
     }
 
     public Line save(Line line) {
+        validateTitle(line);
         return lineRepository.save(line);
     }
 
-    public List<Line> showLines() {
-        return lineRepository.findAll();
+    public void validateTitle(Line line) {
+        lineRepository.findByTitle(line.getTitle())
+                .ifPresent(x -> {
+                    throw new IllegalArgumentException(ERROR_MESSAGE_NAME_OVER_LAP);
+                });
     }
 
     public void updateLine(Long id, Line line) {
-        Line persistLine = lineRepository.findById(id).orElseThrow(RuntimeException::new);
+        validateTitleWhenUpdateInfo(id, line);
+        Line persistLine = lineRepository.findById(id).orElseThrow(()->{
+            throw new IllegalArgumentException(ERROR_MESSAGE_NAME_OVER_LAP);
+        });
```

Comment:
```suggestion
                Line persistLine = lineRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException(ERROR_MESSAGE_NAME_OVER_LAP));
```
throw가 빠지고 위처럼 되어야 할 것 같네요.